app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java

```diff
@@ -41,32 +37,43 @@
     @BindView(R.id.contributionProgress) ProgressBar progressView;
     @BindView(R.id.failed_image_options) LinearLayout failedImageOptions;
 
-    @Inject
-    MediaDataExtractor mediaDataExtractor;
-
-    @Inject
-    @Named("thumbnail-cache")
-    LruCache<String, String> thumbnailCache;
-
-    private DisplayableContribution contribution;
-    private CompositeDisposable compositeDisposable = new CompositeDisposable();
     private int position;
+    private Contribution contribution;
+    private Random random =new Random();
+    private HashMap<Integer,ColorDrawable> drawablesCache=new HashMap();
 
     ContributionViewHolder(View parent, Callback callback) {
         super(parent);
         ButterKnife.bind(this, parent);
         this.callback=callback;
+        populateDrawableCache();
+    }
+    
+    /**
+     * Cache's the drawables to be shown as image placeholders
+     */
+    private void populateDrawableCache() {
```

Comment:
Do we need a cache? was there performance issues in your testing?

I don't think ColorDrawables are that expensive but android can always surprise ya.

Reply:
I did not see a performance issue to be honest but creating new Drawables in onBind, imagine it for 1000 contributions, I thought cache should be okay, anyway this one won't harm, its a small one with 5 entries

Reply:
I am pretty sure android already creates thousands of ColorDrawables,  whenever you set the color of a view. 

It is not like this is a bimap, I think this might be over-engineering

Reply:
Can say that, but do you really think caching like 5 drawables would harm?

Reply:
I think it is unnecessary, If we were to go around doing unnecessary things that do no harm then the code would be littered and the intention that much harder to read. 
Solving problems that exist is plenty hard, no need to invent ones that don't exist.

Reply:
I doubt this hinders the readability, I am okay to remove this, I just don't see a point like this should be the reason. Anyways if you anyways insist so much, I will do it.

Reply:
It hinders the readability, it adds an extra mental burden to the reader and if someone was to come along and see this cache implemented they would rightly assume these are expensive objects and great care must be used when handling them.
The best code is written with the backspace key.

Reply:
Done :)